src/components/TrailList.js

```diff
@@ -18,84 +18,96 @@ const CustomTableCell = withStyles(theme => ({
 }))(TableCell);
 
 class TrailList extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      editableTrail: null,
-    };
-  }
 
   renderTrail = (trail) => {
-    const { selected, trailSelected, modifyTrail, hydrants } = this.props;
-    const { editableTrail } = this.state;
+    const { selected, trailSelected, modifyTrail, hydrants, trailEditable, editableTrail } = this.props;
 
     const id = trail.id;
     const isSelected = selected === id;
     const isEditable = editableTrail === id;
     const rowClass = isSelected ? 'selected' : '';
     const iconClass = `fa fa-${isSelected ? 'minus' : 'plus'}`;
 
+    const toggleHighlight = () => {
+      _.each(trail.features, (f)=> {
+        if (f.get('highlighted')){
+          f.unset('highlighted')
+        } else {
+           f.set('highlighted', true)
+        }
+        f.changed()
+      })
+
+    }
+
+    const handleSelected = () => {
+        isSelected ? null : trailSelected(id)
```

Comment:
It does..but having some issues w/ event bubbling i.e if instead of null I used  trailSelected(null) then when I mean to click on edit and enter the edit form it will deselect the trail instead. Advice?

Reply:
I'll let you decide on the desired behavior, maybe we don't want to un-select on click and let them drop orphans, or we have some other way of doing it, but you can stop bubbling on the edit click event with `e.stopPropogation()`.